value/value.go

```diff
@@ -182,32 +127,325 @@ func (f FieldList) Less(rhs FieldList) bool {
 	}
 }
 
-// List is a list of items.
-type List struct {
-	Items []Value
+type Value interface {
+	IsMap() bool
+	IsList() bool
+	IsBool() bool
+	IsInt() bool
+	IsFloat() bool
+	IsString() bool
+	IsNull() bool
+
+	Map() Map
+	List() List
+	Bool() bool
+	Int() int64
+	Float() float64
+	String() string
+
+	Interface() interface{}
 }
 
-// Less compares two lists lexically.
-func (l *List) Less(rhs *List) bool {
+type List interface {
+	Interface() []interface{}
+	Length() int
+	Iterate(func(int, Value))
+	At(int) Value
+}
+
+type ValueInterface struct {
+	Value interface{}
+}
+
+func (v ValueInterface) IsMap() bool {
+	if _, ok := v.Value.(map[string]interface{}); ok {
+		return true
+	}
+	if _, ok := v.Value.(map[interface{}]interface{}); ok {
+		return true
+	}
+	return false
+}
+
+func (v ValueInterface) Map() Map {
+	if v.Value == nil {
+		return MapString(nil)
+	}
+	switch t := v.Value.(type) {
```

Comment:
I would not expect this?

Reply:
Didn't this already exist?

Reply:
ah, ok, I think I understand now.

Reply:
Oh I was only looking at the last commit